### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by Itswali**
### This PR includes the following:

-     Describe the SOLID principles of OOP. 
-     Refactor your own Ruby code after gaining new knowledge.
-     Apply Ruby best practices and language style guides in code.

`Thanks For the Review.

### CODE REVIEW

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Itswali, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has no contribution from both coding partners, please ensure your partner makes some commits as well. Also kindly make your readme file more descriptive by finishing incomplete sections such as the `Getting Started` section, and the `Authors` section.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+  def create_rental(book_index, person_index, d_date)
+    book = @books[book_index]
+    person = @people[person_index]
+    date = d_date
+    @rentals << Rental.new(date, book, person)
+    puts 'Rental Created.'
+  end
+
+  def list_rentals(_id)
```
Comment (Ismailco) [inline app.rb]: - [ ] Great work on the app, however, the app breaks when trying to list the rentals, kindly make sure to fix this so the user can list the available rentals

- the error 👇🏼 
Screenshot_20230320_194039

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Itswali, @Juanpa8830, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ follow DRY, Solid principles
✔️ Clean code
✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by addod19 — APPROVED** [review]
### Hi @Itswali and @Juanpa8830, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Nice job fixing the bug :rocket: 
- [ ] Nice collaboration :man_technologist: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**